hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutFromLoadAllOperation.java

```diff
@@ -70,8 +71,12 @@ public void run() throws Exception {
             Object previousValue = recordStore.putFromLoad(key, value);
 
             callAfterPutInterceptors(value);
+            // Null-pointer here, in case of null key loaded by MapLoader
+            checkNotNull(key, "Key loaded by a MapLoader cannot be null.");
```

Comment:
@tombujok OK but if value is null, we're still calling interceptor. I believe we're not saving null values to Map, right ?

Also, as you said, `recordStore.putFromLoad(key, value);` check for both key & value. In this case, even value is null, we're still calling interceptor, just precent NPE inside if block, raise event etc.